### PR TITLE
Fix/KFS-1609 multibyte chars

### PR DESCRIPTION
**Pull request by gustavodemorais**
I fixed the util functions that we use and will create a ticket to fix the other ones. More details in the ticket.

### CODE REVIEW

File: internal/runes/runes.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package runes
+
+import "github.com/samber/lo"
+
+// IndexNotByte is similar with strings.IndexByte but showing the opposite behavior.
+func IndexOfNot(x []rune, c byte) int {
+	i := lo.LastIndexOf(x, ' ')
```
Comment (Airblader) [inline internal/runes/runes.go]: Should this be `LastIndexOf(x, c)`? The `c` is unused currently, right?

File: internal/runes/runes.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package runes
+
+import "github.com/samber/lo"
+
+// IndexNotByte is similar with strings.IndexByte but showing the opposite behavior.
```
Comment (Airblader) [inline internal/runes/runes.go]: I would find it easier to just describe the actual behavior. I'm not entirely sure what to expect from "opposite of index-of", but I'd go with "find the first index that is _not_ `c`" -- I can't map this understanding to the implementation, though.

File: internal/runes/runes.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+func IndexOfNot(x []rune, c byte) int {
+	i := lo.LastIndexOf(x, ' ')
+
+	if i == -1 && i+1 < len(x) {
```
Comment (Airblader) [inline internal/runes/runes.go]: This condition seems rather odd. If `i == -1`, why not just write `&& len(x) > 0` and `return 0`?

File: internal/runes/runes.go [hunk 4]
```diff
@@ -1,22 +1,10 @@
...
-
-	return -1
-}
-
-// LastIndexNotByte is similar with strings.LastIndexByte but showing the opposite behavior.
-func LastIndexNotByte(s string, c byte) int {
-	for i := len(s) - 1; i >= 0; i-- {
-		if s[i] != c {
+// IndexOfNot returns the index of the first rune in x that is not c
+func IndexOfNot(x []rune, c rune) int {
```
Comment (Airblader) [inline internal/runes/runes.go:4]: Well, I should not have reviewed commit-by-commit, because now it all makes more sense 🍭

**Review by Airblader — APPROVED** [review]
LGTM since all my comments resolved themselves in the next commit. Personally, seems overkill to add Godash (Lodash for Go, get it? Why the heck did they not use that name?!) for the _one_ remaining usage (afaict), though.

File: internal/runes/runes.go (hunk 1 above)
Comment (Airblader) [reply internal/runes/runes.go]: Resolved in next commit

File: internal/runes/runes.go (hunk 2 above)
Comment (Airblader) [reply internal/runes/runes.go]: Resolved in next commit

File: internal/runes/runes.go (hunk 3 above)
Comment (Airblader) [reply internal/runes/runes.go]: Resolved in next commit

File: internal/runes/runes.go (hunk 2 above)
Comment (gustavodemorais) [reply internal/runes/runes.go]: Yeah, this is a nonsense function description lol. This was copied from go-prompt's similar implementation but I updated it

**Comment by gustavodemorais** [comment]
> LGTM since all my comments resolved themselves in the next commit. Personally, seems overkill to add Godash (Lodash for Go, get it? Why the heck did they not use that name?!) for the _one_ remaining usage (afaict), though.

Coming from the frontend world, I totally agree that loading a lib like loadash only for a one func is totally using a sledgehammer to crack a nut. 

However, the beauty in go is that unused stuff from a package will be dropped by the linker and won't clutter the binary, so a library like this makes much more sense in go. 

Also, I think at least in all repos I worked in our team(cli, cc-ksql-service, language service) we informally try to use this (quite nice) lib when we can to reduce the bloat from go.

File: internal/runes/runes.go (hunk 4 above)
Comment (gustavodemorais) [reply internal/runes/runes.go:4]: Yeah, sorry! I also like commit-by-commit PRs, but they are generally easier to create for features I know I have a clear path in mind. In the case of this bug here, it was a bit more messy.

**Pull request merged**